### PR TITLE
Koalas Series.str functions

### PR DESCRIPTION
**Pull request by hjoo**
As discussed in #281, this PR creates Series.str functions for manipulating strings in Koalas Series.

### CODE REVIEW

**Comment by hjoo** [comment]
Pyspark 2.3 doesn't support BinaryType so `Series.str.encode()` and `Series.str.decode()` fail.

Would we prefer to:
1)  remove support for `encode()` and `decode()`?
2) check pyspark versions in the method implementations and disable unit tests if pyspark version < 2.4?

If we go the second option, is there a prior way of skipping unit tests based on a package version?

**Comment by icexelloss** [comment]
Per the purpose of this PR I'd prefer to leave out `encode` and `decode` because I think binary type needs to be addressed separately. @HyukjinKwon WDYT?

**Comment by ueshin** [comment]
Yes, we can remove the two functions from here and address 2) in a separate PR.
In that case, please leave a comment saying we'll add them later on #281 and go ahead.
Thanks!

File: databricks/koalas/strings.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+    def lower(self) -> 'ks.Series':
+        """
+        Convert strings in the Series/Index to lowercase.
+        """
+        return _wrap_accessor_pandas(
+            self,
+            lambda x: x.str.lower(),
+            StringType()
+        ).alias(self.name)
```
[the remark this answers is not in the document]
Comment (hjoo) [reply databricks/koalas/strings.py]: Yes, I'll add `upper()` to this PR.

**Comment by hjoo** [comment]
I've updated the code to include `upper()` and the strip functions: `strip()`, `lstrip()`, `rstrip()`.
I'll update #281 with a list of the `Series.str` functions that are still left.

**Comment by ueshin** [comment]
Thanks! merging.

**Comment by HyukjinKwon** [comment]
nice!

**Pull request merged**